applications/welcome/models/access_control.py

```diff
@@ -0,0 +1,61 @@
+#########################################################################
+## This file defines access control methods, including verification, setting and deleting.
+## - access_view can view only.
+## - access_setting can change settings, and view variables/output, but cannot change procedure code.
+## - access_manage can do everything the main user can do.
+#########################################################################
+
```

Comment:
I think we need some comments here on the device_token + "_" + user_id construct you use.  I guess you take this to be a unique identifier of the device, because the device_token might not be unique across all users?   But no, because the user_id here is the user_id of the user who wants to view/edit, not the user_id of the user who owns the device.  Can you clarify what is that?